internal/cmd/export.go

```diff
@@ -1078,6 +1084,101 @@ func gatherPatroniInfo(ctx context.Context,
 	return nil
 }
 
+// gatherSystemTime takes a client and buffer and collects system time
+// in each Pod and calculates the delta against client system time.
+func gatherSystemTime(ctx context.Context,
+	clientset *kubernetes.Clientset,
+	config *rest.Config,
+	namespace string,
+	clusterName string,
+	tw *tar.Writer,
+	cmd *cobra.Command,
+) error {
+	writeInfo(cmd, "Collecting system times from containers...")
+	// Get the cluster Pods by label
+	pods, err := clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: "postgres-operator.crunchydata.com/cluster=" + clusterName,
+	})
+	if err != nil {
+		if apierrors.IsForbidden(err) {
+			writeInfo(cmd, err.Error())
+			return nil
+		}
+		return err
+	}
+
+	if len(pods.Items) == 0 {
+		// If we didn't find any resources, skip
+		writeInfo(cmd, "PostgresCluster Pods not found when gathering system time information, skipping")
+		return nil
+	}
+
+	podExec, err := util.NewPodExecutor(config)
+	if err != nil {
+		return err
+	}
+
+	var buf bytes.Buffer
+	for _, pod := range pods.Items {
+		for _, container := range pod.Spec.Containers {
+			// Attempt to exec in and run 'date' command in all available containers,
+			// regardless of state, etc.
+			exec := func(stdin io.Reader, stdout, stderr io.Writer, command ...string,
+			) error {
+				return podExec(namespace, pod.GetName(), container.Name,
+					stdin, stdout, stderr, command...)
+			}
+
+			stdout, stderr, err := Executor(exec).systemTime()
+			if err != nil {
+				// If we get an RBAC error, let the user know. Otherwise, just
+				// try the next container.
+				if apierrors.IsForbidden(err) {
+					writeInfo(cmd, fmt.Sprintf(
+						"Failed to get system time for Container \"%s\" in Pod \"%s\". Error: \"%s\"",
+						container.Name, pod.GetName(), err.Error()))
+				}
+				continue
+			}
+
+			// Get client datetime.
+			clientDateTimeStr := time.Now().UTC().Format(time.UnixDate)
+			// Normalize client time for calculating delta.
+			clientTime, err := time.Parse(time.UnixDate, clientDateTimeStr)
+			if err != nil {
+				return err
+			}
+
+			// Get container datetime.
+			containerDateTime, err := time.Parse(time.UnixDate, strings.TrimSpace(stdout))
+			containerDateTimeStr := containerDateTime.Format(time.UnixDate)
+			if err != nil {
+				return err
+			}
+
+			// Calculate difference between client and container datetime.
+			delta := clientTime.Sub(containerDateTime)
+			deltaStr := fmt.Sprintln(delta)
+
+			// Build report.
+			buf.Write([]byte("Pod: " + pod.GetName() + "\t" + "Container: " + container.Name + "\n"))
+			buf.Write([]byte("\tClient time: " + clientDateTimeStr + "\n"))
+			buf.Write([]byte("\tContainer time: " + containerDateTimeStr + "\n"))
```

Comment:
⛏️ [fmt.Fprint](https://pkg.go.dev/fmt#Fprint) and friends are another way to write to a buffer.

```suggestion
			fmt.Fprintln(buf, "Pod: " + pod.GetName() + "\t" + "Container: " + container.Name))
			fmt.Fprintln(buf, "\tClient time: " + clientDateTimeStr))
			fmt.Fprintln(buf, "\tContainer time: " + containerDateTimeStr))
```

Reply:
```suggestion
			fmt.Fprintln(buf, "Pod: " + pod.GetName() + "\t" + "Container: " + container.Name + 
				"\tClient time: " + clientDateTimeStr + "\tContainer time: " + containerDateTimeStr)
```

Reply:
@benjaminjb Do you want everything on one line?

Reply:
This format might be easy to see what you're looking for at a glance:
```
Delta: 0s		 Pod name: hippo-instance1-9xl8-0
Pod time:		 Wed Nov  1 14:42:33 UTC 2023
Client time:		 Wed Nov  1 14:42:33 UTC 2023
```

Reply:
Just to verify, this is what the committed code currently does
```
Pod: hippo-instance1-9xl8-0   Container: database
    Client time: Wed Nov  1 14:42:33 UTC 2023
    Container time: Wed Nov  1 14:42:33 UTC 2023
    Delta: 0s
```
That's pretty clear about one pod's time, but I guess I was thinking that to compare times across pods, it might be easier if it were
```
Pod: hippo-instance1-9xl8-0   Container: database    Client time: Wed Nov  1 14:42:33 UTC 2023    Container time: Wed Nov  1 14:42:33 UTC 2023   Delta: 0s
Pod: hippo-repo-0   Container: pgbackrest    Client time: Wed Nov  1 14:42:33 UTC 2023    Container time: Wed Nov  1 14:42:33 UTC 2023   Delta: 0s
```
Of course, with pod and container being variable, the lines won't line up unless we pad or put them at the end (potentially time might also end up like that).

Reply:
That's correct about what's currently committed. And, good point about comparing pod times. Insofar as what you want to know first is temporal diffs,  maybe we should just lead with the times?
```
Delta: 0s	Pod time: Wed Nov  1 15:12:57 UTC 2023	Client time: Wed Nov  1 15:12:57 UTC 2023	Pod name: hippo-instance1-9xl8-0
Delta: 0s	Pod time: Wed Nov  1 15:12:57 UTC 2023	Client time: Wed Nov  1 15:12:57 UTC 2023	Pod name: hippo-instance1-v4jw-0
Delta: 0s	Pod time: Wed Nov  1 15:12:58 UTC 2023	Client time: Wed Nov  1 15:12:58 UTC 2023	Pod name: hippo-pgadmin-0
Delta: 0s	Pod time: Wed Nov  1 15:12:58 UTC 2023	Client time: Wed Nov  1 15:12:58 UTC 2023	Pod name: hippo-pgbouncer-6cc64db4d7-8prv7
Delta: 0s	Pod time: Wed Nov  1 15:12:59 UTC 2023	Client time: Wed Nov  1 15:12:59 UTC 2023	Pod name: hippo-repo-host-0
```